src/gui/layout/qgslayoutruler.cpp

```diff
@@ -802,6 +802,8 @@ void QgsLayoutRuler::mouseReleaseEvent( QMouseEvent *event )
   }
   else if ( event->button() == Qt::RightButton )
   {
+    QMouseEvent leftReleaseEvent( event->type(), event->localPos(), Qt::LeftButton, event->buttons(), event->modifiers() );
+    mouseReleaseEvent( &leftReleaseEvent );
```

Comment:
This change looks dangerous to me -- it will result in a second left mouse button release event being sent to the rulers, and looking at QgsLayoutRuler::mouseReleaseEvent this second event will trigger a guide creation event. I think more logic is required in mousePressEvent/mouseReleaseEvent to correctly discard an unexpected release event whichs occurs outside of the usual press/release event pairs.

Reply:
@nyalldawson Yes, I didn't know which should be the expected behavior, as the right click here is probably a mistake from the user, I thought that we could let the guide being created or moved. Then the user can move it yet, if necessary.

But ok, discard is probably the cleaner way to handle this. I changed it : I had to add a new attribute to QgsLayoutRuler, to keep the old position of an existing guide. This attribute is used to replace the guide to the previous position.

Reply:
I also change to logic in the select tool to discard the action and to keep the current selection.